NorthstarDLL/core/filesystem/rpakfilesystem.cpp

```diff
@@ -255,11 +255,20 @@ void*, __fastcall, (int nPakHandle, void* pCallback))
 
 // we hook this exclusively for resolving stbsp paths, but seemingly it's also used for other stuff like vpk, rpak, mprj and starpak loads
 // tbh this actually might be for memory mapped files or something, would make sense i think
+// rtech_game.dll + 0x1e20
 // clang-format off
 HOOK(ReadFileAsyncHook, ReadFileAsync, 
 void*, __fastcall, (const char* pPath, void* pCallback))
 // clang-format on
 {
+	// NOTE [Fifty]: For some reason some users are getting pPath as null when
+	//               loading a server, ReadFileAsync uses CreateFileA and checks
+	//               its return value so this is completely safe
+	if (pPath == NULL)
```

Comment:
There may be other possible path values we want to skip over

```c
if (IsBadStringPtrA(pPath, -1))
```
should do the trick

Reply:
> **Important:** This function is obsolete and should not be used. Despite its name, it does not guarantee that the pointer is valid or that the memory pointed to is safe to use. For more information, see Remarks on this page.

Reply:
I suggested it because of its use within the wine codebase, which appears to have a reliable implementation of it, but it seems less reliable on Windows.

Here is a post by Raymond Chen about it from 2006 https://devblogs.microsoft.com/oldnewthing/20060927-07/?p=29563